src/arrays.jl

```diff
@@ -45,4 +48,45 @@ function \(P::ArrayPlan, f::AbstractArray)
     fr = reshape(fp, size(fp,1), prod(size(fp)[2:end]))
 
     permutedims(reshape(F\fr, size(fp)...), inv_perm(perm))
+end
+
+struct NDimsPlan{T, FF<:ArrayPlan{<:T}, Dims<:Tuple} <: Plan{T}
+    F::FF
+    dims::Dims
+    function NDimsPlan(F, dims)
+        if length(Set(size(F, dims...))) > 1
+            error("Different size in dims axes not yet implemented in N-dimensional transform.")
+        end
+        new{eltype(F), typeof(F), typeof(dims)}(F, dims)
+    end
+end
+
+size(P::NDimsPlan) = size(P.F)
+size(P::NDimsPlan, k::Int) = size(P.F, k)
+size(P::NDimsPlan, k...) = size(P.F, k...)
+
+function *(P::NDimsPlan, f::AbstractArray)
+    F, dims = P.F, P.dims
+    @assert size(F) == size(f)
+    g = copy(f)
+    t = 1:ndims(g)
+    for d in dims
+        perm = ntuple(k -> k == 1 ? t[d] : k == d ? t[1] : t[k], length(t))

```

Comment:
```suggestion
        perm = ntuple(k -> k == 1 ? t[d] : k == d ? t[1] : t[k], ndims(g))
```
This might be easier on type-inference